src/Cli/dotnet/CommonOptions.cs

```diff
@@ -100,6 +115,38 @@ public static bool VerbosityIsDetailedOrDiagnostic(this VerbosityOptions verbosi
                 verbosity.Equals(VerbosityOptions.d) ||
                 verbosity.Equals(VerbosityOptions.detailed);
         }
+
+        internal static IEnumerable<string> ResolveArchOptionToRuntimeIdentifier(string arg)
+        {
+            var currentRid = GetCurrentRuntimeId();
+            var os = currentRid.Substring(0, currentRid.LastIndexOf("-"));
+            return new string[] { $"-property:RuntimeIdentifier={os}-{arg}", "-property:SelfContained=true" };
```

Comment:
Per my read, the default when using the new shorthand combinations is `--selfcontained true`. That's not the intention, but the opposite.

So, two things:

- The default should the `--self-contained false`
- It should be possible for someone to set `--self-contained true`.

For example, the first of these should be framework-dependent and the other self-contained:

- `dotnet build -a x64`
- `dotnet build -a x64 --self-contained`